### PR TITLE
chore: add semantic commit check

### PR DESCRIPTION
**Pull request by dsanders11**
We've been using semantic commit messages for this project, let's enforce it.

### CODE REVIEW

File: .github/workflows/semantic-commits.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      - edited
+      - synchronize
+
+jobs:
+  main:
+    name: Validate PR Title
+    runs-on: ubuntu-latest
+    steps:
+      - name: semantic-pull-request
+        uses: amannn/action-semantic-pull-request@91682d0
```
Comment (RaisinTen) [inline .github/workflows/semantic-commits.yml:16]:
```suggestion
        uses: amannn/action-semantic-pull-request@@v4
```
Why not a version instead?

**Review by RaisinTen — APPROVED** [review]
LGTM

**Comment by dsanders11** [comment]
> Actually, one more question - does this allow commits that start with [skip ci] like in https://github.com/postmanlabs/postject/pull/28?

I believe you would just need to put the `[skip ci]` somewhere else other than the start of the message, like on the end instead, which I think is preferable anyway since it keeps the starts of messages clean and easy to scan.

File: .github/workflows/semantic-commits.yml (hunk 1 above)
Comment (dsanders11) [reply .github/workflows/semantic-commits.yml]: It's considered [a security best practice by GitHub](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) when using third-party actions:

> Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload.

Prevents a future supply chain attack if someone were to compromise that repo and change the tag to point at malicious code.

Added benefit of also ensuring the check doesn't break without us doing anything if the author inadvertently ships a buggy version.

**Pull request merged**